lib/webstorage.service.spec.ts

```diff
@@ -0,0 +1,257 @@
+
+			webStorageService.reInit();
+
+			sinon.assert.calledOnce(mockCheckSupport);
+			expect(isSupported).to.be.equal(isSupported);
+
+		})
+	});
+
+	describe('derive key', () => {
+		it('should handle empty user strings', () => {
+			const key: string = "ThisIsAKey";
+			const expectedKey = `ls.${key}`;
+
+			expect(webStorageService.deriveKey(key)).to.be.equal(expectedKey);
+		});
+
+		it('should handle user strings', () => {
+			const key: string = "ThisIsAKey";
+			const user: string = "UserName";
+			const expectedKey = `ls.${user}.${key}`;
+
+			expect(webStorageService.deriveKey(key, user)).to.be.equal(expectedKey);
+		});
+	});
+
+	describe('get', () => {
+		let mockBrowserStorage: Storage;
+		let fakeItem = { Id: 123 };
+		let warnSpy: sinon.SinonSpy;
+		let getItemSpy: sinon.SinonSpy;
+
+		beforeEach(() => {
+			warnSpy = sinon.spy(() => console.warn);
+			console = <any>{
+				log: console.log,
+				error: console.error,
+				warn: warnSpy
+			};
+
+			getItemSpy = sinon.spy(() => JSON.stringify(fakeItem));
+			mockBrowserStorage = <any>{ getItem: <any>getItemSpy, mockBrowserStorage: mockBrowserStorage };
+			webStorageService = new WebStorageService({ ...defaultConfig, mockWebStorage: mockBrowserStorage });
+		});
+
+		it('should check if web storage is supported by the browser', () => {
+			const isSupported: boolean = true;
+			let mockCheckSupport = sinon.spy(() => isSupported);
+			webStorageService.checkSupport = mockCheckSupport
+
+			webStorageService.reInit();
+
+			webStorageService.get('dontCare');
+
+			sinon.assert.notCalled(warnSpy);
+		});
+
+		it('should display console log when web storage is not supported', () => {
+			const isSupported: boolean = false;
+			let mockCheckSupport = sinon.spy(() => isSupported);
+			webStorageService.checkSupport = mockCheckSupport
+
+			webStorageService.reInit();
+
+			webStorageService.get('dontCare');
+
+			sinon.assert.calledOnce(warnSpy);
+			sinon.assert.calledWith(warnSpy, 'webStorageError', WEB_STORAGE_NOT_SUPPORTED);
+		});
+
+		it('should make call to getItem to retreive data from web storage', () => {
+			const isSupported: boolean = true;
+			let mockCheckSupport = sinon.spy(() => isSupported);
+			webStorageService.checkSupport = mockCheckSupport
+
+			webStorageService.reInit();
+
+			webStorageService.get('dontCare');
+
+			sinon.assert.calledOnce(getItemSpy);
+		});
+
+		it('should parse json string to real object', () => {
+			const isSupported: boolean = true;
+			let mockCheckSupport = sinon.spy(() => isSupported);
+			webStorageService.checkSupport = mockCheckSupport
+
+			webStorageService.reInit();
+
+			let result = webStorageService.get('dontCare');
+
+			expect(result).to.deep.equal(fakeItem);
+		});
+	});
+
+	describe('set', () => {
+		let mockBrowserStorage: Storage;
+		let fakeItem = { Id: 123 };
+		let warnSpy: sinon.SinonSpy;
+		let setItemSpy: sinon.SinonSpy;
+
+		beforeEach(() => {
+			warnSpy = sinon.spy(() => console.warn);
```

Comment:
It is possible to spy on the original console.warn function without replacing the entire console object, i.e. `warnSpy = sinon.spy(console, 'warn');`